components/eventing-controller/pkg/ems/httpclient/client.go

```diff
@@ -19,7 +20,7 @@ type Client struct {
 }
 
 func NewHttpClient(cfg clientcredentials.Config) *Client {
-	ctx := signals.NewContext()
+	ctx := signals.NewReusableContext()
```

Comment:
Redesign to use only one reusable interruptible context instead to spread such contexts (cancelable++) through the code.